ExtractUserDataActivity/handler.ts

```diff
@@ -0,0 +1,526 @@
+  t.partial({ query: t.string })
+]);
+export type ActivityResultQueryFailure = t.TypeOf<
+  typeof ActivityResultQueryFailure
+>;
+
+// activity failed for user not found
+const ActivityResultUserNotFound = t.interface({
+  kind: t.literal("USER_NOT_FOUND_FAILURE")
+});
+export type ActivityResultUserNotFound = t.TypeOf<
+  typeof ActivityResultUserNotFound
+>;
+
+export const ActivityResultFailure = t.taggedUnion("kind", [
+  ActivityResultUserNotFound,
+  ActivityResultQueryFailure,
+  ActivityResultInvalidInputFailure
+]);
+export type ActivityResultFailure = t.TypeOf<typeof ActivityResultFailure>;
+
+export const ActivityResult = t.taggedUnion("kind", [
+  ActivityResultSuccess,
+  ActivityResultFailure
+]);
+export type ActivityResult = t.TypeOf<typeof ActivityResult>;
+
+const logPrefix = `ExtractUserDataActivity`;
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ * @param queryName an optional name for the query, for logging purpose
+ *
+ * @returns either the query result or a query failure
+ */
+const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<QueryError | Error, R>>,
+  queryName: string = ""
+): TaskEither<ActivityResultQueryFailure, R> =>
+  tryCatch(lazyPromise, (err: Error) =>
+    ActivityResultQueryFailure.encode({
+      kind: "QUERY_FAILURE",
+      query: queryName,
+      reason: err.message
+    })
+  ).chain((queryErrorOrRecord: Either<QueryError | Error, R>) =>
+    fromEither(
+      queryErrorOrRecord.mapLeft(queryError =>
+        ActivityResultQueryFailure.encode({
+          kind: "QUERY_FAILURE",
+          query: queryName,
+          reason: JSON.stringify(queryError)
+        })
+      )
+    )
+  );
+
+/**
+ * To be used for exhaustive checks
+ * @param _
+ */
+function assertNever(_: never): void {
+  throw new Error("should not have executed this");
+}
+
+/**
+ * Logs depending on failure type
+ * @param context the Azure functions context
+ * @param failure the failure to log
+ */
+const logFailure = (context: Context) => (
+  failure: ActivityResultFailure
+): void => {
+  switch (failure.kind) {
+    case "INVALID_INPUT_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error decoding input|ERROR=${failure.reason}`
+      );
+      break;
+    case "QUERY_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error ${failure.query} query error |ERROR=${failure.reason}`
+      );
+      break;
+    case "USER_NOT_FOUND_FAILURE":
+      context.log.error(`${logPrefix}|Error user not found |ERROR=`);
+      break;
+    default:
+      assertNever(failure);
+  }
+};
+
+/**
+ * Skims a db document from db-related fields
+ * @param doc the document as retrieved from db
+ *
+ * @returns the same document without db metadata
+ */
+const fromRetrievedDbDocument = <T>(doc: T & RetrievedDocumentT): T => {
```

Comment:
every model has a class method `toBaseType` method which does exactly the same and does not depend on cosmosdb internal, consider using that one and remove this one
ie. https://github.com/pagopa/io-functions-commons/blob/6b60b5cc5fb840ff397f2ebc3261a3c7408b2168/src/models/profile.ts#L129

Reply:
it seems to to be used by anyone. It's passed to the base class which exposes it as protected, then never actually used.
https://github.com/pagopa/io-functions-commons/blob/master/src/utils/documentdb_model.ts#L31

Reply:
If we cannot use that, I suggest to remove this method anyway and just use 
`t.exact(Message).encode(o);` inline when needed